openedx/core/djangoapps/profile_images/images.py

```diff
@@ -127,51 +120,102 @@ def validate_uploaded_image(uploaded_file):
     uploaded_file.seek(0)
 
 
-def _get_scaled_image_file(image_obj, size):
+def _crop_image_to_square(image):
+    """
+    Given a PIL.Image object, return a copy cropped to a square around the
+    center point with each side set to the size of the smaller dimension.
+    """
+    width, height = image.size
+    if width != height:
+        side = width if width < height else height
+        left = (width - side) // 2
+        top = (height - side) // 2
+        right = (width + side) // 2
+        bottom = (height + side) // 2
+        image = image.crop((left, top, right, bottom))
+    return image
+
+
+def _set_color_mode_to_rgb(image):
+    """
+    Given a PIL.Image object, return a copy with the color mode set to RGB.
+    """
+    return image.convert('RGB')
+
+
+def _scale_image(image, size):
+    """
+    Given a PIL.Image object, get a resized copy with each side being `size`
+    pixels long.  The scaled image will always be square.
+    """
+    return image.resize((size, size), Image.ANTIALIAS)
+
+
+def _create_image_file(image_obj, exif):
     """
-    Given a PIL.Image object, get a resized copy using `size` (square) and
-    return a file-like object containing the data saved as a JPEG.
+    Given a PIL.Image object, and return a file-like object containing the data
+    saved as a JPEG.
 
-    Note that the file object returned is a django ContentFile which holds
-    data in memory (not on disk).
+    Note that the file object returned is a django ContentFile which holds data
+    in memory (not on disk).
     """
-    if image_obj.mode != "RGB":
-        image_obj = image_obj.convert("RGB")
-    scaled = image_obj.resize((size, size), Image.ANTIALIAS)
     string_io = StringIO()
-    scaled.save(string_io, format='JPEG')
+    if exif is None:
+        image_obj.save(string_io, format='JPEG')
+    else:
+        image_obj.save(string_io, format='JPEG', exif=exif)
     image_file = ContentFile(string_io.getvalue())
     return image_file
 
 
-def create_profile_images(image_file, profile_image_names):
+def _get_corrected_exif(image, original):
     """
-    Generates a set of image files based on image_file and
-    stores them according to the sizes and filenames specified
-    in `profile_image_names`.
+    If the original image contains exif data, use that data to
+    preserve image orientation in the new image.
     """
-    image_obj = Image.open(image_file)
+    if 'exif' in original.info:
+        image_exif = image.info.get('exif', piexif.dump({}))
+        original_exif = original.info['exif']
+        image_exif = _update_exif_orientation(image_exif, _get_exif_orientation(original_exif))
+        return image_exif
 
-    # first center-crop the image if needed (but no scaling yet).
-    width, height = image_obj.size
-    if width != height:
-        side = width if width < height else height
-        image_obj = image_obj.crop(((width - side) / 2, (height - side) / 2, (width + side) / 2, (height + side) / 2))
```

Comment:
:+1: I like the updates you made to this "center-cropping" routine, much easier to read at a glance now